package/android/cpp/rnskia-android/RNSkOpenGLCanvasProvider.cpp

```diff
@@ -39,6 +39,7 @@ bool RNSkOpenGLCanvasProvider::renderToCanvas(
       if (!_surfaceHolder->makeCurrent()) {
         return false;
       }
+      _surfaceHolder->updateTexImage();
```

Comment:
I would expect this one to be called after line 45, otherwise you'll be pushing the previous frame to screen?

Reply:
that's correct but we need to release from the UI thread loop for the change to happen (will double check) this is also why we now have renderImmediate on surfaceChange. I will check if we can sanitize things here.

Reply:
I even tried things like:
      auto result = _surfaceHolder->present();
      glFinish();
      _surfaceHolder->updateTexImage();
      return result;

But since we are blocking the thread, we are not letting the surface texture to swap it's hardware buffer